base/broadcast.jl

```diff
@@ -926,13 +926,12 @@ function copyto_nonleaf!(dest, bc::Broadcasted, iter, state, count)
         y === nothing && break
         I, state = y
         @inbounds val = bc[I]
-        S = typeof(val)
-        if S <: T
+        if val isa T || typeof(val) === T
```

Comment:
Whether `typeof(val) === T` is needed is not clear (https://github.com/JuliaLang/julia/issues/30125), but I've added it for consistency with `collect_to_with_first!`. If that's redundant, we should remove all uses at the same time.